CODING_STYLE.md

```diff
@@ -0,0 +1,172 @@
+2. Private, non-static, non-const fields prefixed with m_.
+3. Avoid public fields, except in structs.
+4. Use `override`, `final` and `const` as much as possible.
+5. No implementations with the class declaration, except:
+   - template or force-inline method (though prefer implementation at bottom of header file).
+   - one-line implementation (in which case include it in same line as declaration).
+6. For a property `foo`
+   - Member: `m_foo`;
+   - Getter: `foo()`; also: for booleans, `isFoo()`
+   - Setter: `setFoo()`;
+
+
+## Naming
+
+1. Collection conventions:
+   - `...s` means `std::vector` e.g. `using MyTypes = std::vector<MyType>`
+   - `...Set` means `std::set` e.g. `using MyTypeSet = std::set<MyType>`
+   - `...Hash` means `std::unordered_set` e.g. `using MyTypeHash = std::unordered_set<MyType>`
+2. Class conventions:
+   - `...Face` means the interface of some shared concept. (e.g. `FooFace` might be a pure virtual class.)
+3. Avoid unpronounceable names:
+   - If you need to shorten a name favour a pronouncable slice of the original to a scattered set of consonants.
+   - e.g. `Manager` shortens to `Man` rather than `Mgr`.
+4. Avoid prefixes of initials (e.g. DON'T use `IMyInterface`, `CMyImplementation`)
+5. Find short, memorable & (at least semi-) descriptive names for commonly used classes or name-fragments.
+   - A dictionary and thesaurus are your friends.
+   - Spell correctly.
+   - Think carefully about the class's purpose.
+   - Imagine it as an isolated component to try to decontextualise it when considering its name.
+   - Don't be trapped into naming it (purely) in terms of its implementation.
+
+
+
+## Type-definitions
+
+1. Prefer `using` to `typedef`. E.g. `using ints = std::vector<int>` rather than
+   `typedef std::vector<int> ints`.
+2. Generally avoid shortening a standard form that already includes all important information:
+   - e.g. stick to `shared_ptr<X>` rather than shortening to `ptr<X>`.
+3. In general expressions should be roughly as important/semantically meaningful as the space they occupy.
+
+
+
+
+## Logging
+- Use `INFO` for non-critical, informative logging messages
+- Use `WARNING` for logging potentially dangerous messages
+- Use `FATAL` for logging a message and thereafter terminating the process
+
+
+## Recommended reading
+
+Herb Sutter and Bjarne Stroustrup
+- "C++ Core Guidelines" (https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md)
+
+Herb Sutter and Andrei Alexandrescu
+- "C++ Coding Standards: 101 Rules, Guidelines, and Best Practices"
+
+Scott Meyers
+- "Effective C++: 55 Specific Ways to Improve Your Programs and Designs (3rd Edition)"
+- "More Effective C++: 35 New Ways to Improve Your Programs and Designs"
+- "Effective Modern C++: 42 Specific Ways to Improve Your Use of C++11 and C++14"
+
+
+[F.21]: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f21-to-return-multiple-out-values-prefer-returning-a-tuple-or-struct
```

Comment:
I think the anchor on the linked page has changed recently. It should be: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-out-multi